android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountSelectorActivity.java

```diff
@@ -92,4 +70,39 @@ public void onAccountClick(WalletListItemModel walletListItemModel) {
             }
         }
     }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private void initAccounts() {
+        mKeyringModel.getAccounts(accountInfos -> {
+            mAccountInfos = accountInfos;
+            List<WalletListItemModel> walletListItemModelList = new ArrayList<>();
+            for (AccountInfo accountInfo : mAccountInfos) {
+                if (!accountInfo.isImported) {
+                    walletListItemModelList.add(
+                            new WalletListItemModel(R.drawable.ic_eth, accountInfo.name,
+                                    accountInfo.address, null, null, accountInfo.isImported));
+                }
+            }
+
+            mWalletCoinAdapter.setWalletListItemModelList(walletListItemModelList);
+            mWalletCoinAdapter.notifyDataSetChanged();
+
+            mKeyringModel.mSelectedAccount.observe(this, selectedAccountInfo -> {
+                if (selectedAccountInfo == null) return;
+
+                mWalletCoinAdapter.updateSelectedNetwork(
+                        selectedAccountInfo.name, selectedAccountInfo.address);
+                boolean callDataSetChanged = true;
+                for (AccountInfo accountInfo : mAccountInfos) {
+                    if (selectedAccountInfo.address.equals(accountInfo.address)) {
+                        callDataSetChanged = false;
+                        break;
+                    }
+                }
+                if (callDataSetChanged) {
+                    initAccounts();
```

Comment:
recursive call to this method will add another anonymous observer at `mKeyringModel.mSelectedAccount.observe(this, selectedAccountInfo -> {` . if you cannot use the `mAccountAllAccountsPair` then move the `mKeyringModel.getAccounts(accountInfos -> {` call inside the observer of `mSelectedAccount` and use it with the adapter

Reply:
yeah, I also don't like that it's recursive, I re-made it to get accounts inside the observer.